hoot-core/src/main/cpp/hoot/core/Units.h

```diff
@@ -31,15 +31,89 @@
 #define _USE_MATH_DEFINES
 #include <math.h>
 
+//Boost units
+#include <boost/units/systems/si.hpp>
+#include <boost/units/base_units/us/foot.hpp>
+#include <boost/units/base_units/us/mile.hpp>
+#include <boost/units/base_units/metric/knot.hpp>
+#include <boost/units/base_units/metric/nautical_mile.hpp>
+#include <boost/units/static_constant.hpp>
+#include <boost/units/systems/si/time.hpp>
+
 namespace hoot
 {
   typedef double Degrees;
   typedef double Meters;
   typedef double Radians;
 
+  //Boost types
+  typedef boost::units::quantity<boost::units::si::time, double> Times;
+  typedef boost::units::quantity<boost::units::si::length, double> Length;
+  typedef boost::units::quantity<boost::units::si::velocity, double> Velocity;
+
+  typedef boost::units::us::foot_base_unit::unit_type foot_unit;
+  BOOST_UNITS_STATIC_CONSTANT(feet, foot_unit);
+  typedef boost::units::us::mile_base_unit::unit_type mile_unit;
+  BOOST_UNITS_STATIC_CONSTANT(mile, mile_unit);
+  typedef boost::units::metric::nautical_mile_base_unit::unit_type nmi_unit;
+  BOOST_UNITS_STATIC_CONSTANT(nmi, nmi_unit);
+  typedef boost::units::metric::knot_base_unit::unit_type knot_unit;
+  BOOST_UNITS_STATIC_CONSTANT(knot, knot_unit);
+
   inline Radians toRadians(Degrees d) { return d / 180.0 * M_PI; }
 
   inline Degrees toDegrees(Radians r) { return r / M_PI * 180.0; }
+
+  //Boost units functions
+  inline Length kiloToLength(double f)
+  {
+    Length l = 1000.0 * boost::units::si::meters;
+    return static_cast<Length>(f * l);
+  }
+
+  inline Length decimiToLength(double f)
+  {
+    Length l = 0.1 * boost::units::si::meters;
+    return static_cast<Length>(f * l);
+  }
+
+  inline Length feetToLength(double f)
+  {
+    return static_cast<Length>(f * feet);
+  }
+
+  inline Length mileToLength(double f)
+  {
+    return static_cast<Length>(f * mile);
+  }
+
+  inline Length nmiToLength(double f)
+  {
+    return static_cast<Length>(f * nmi);
+  }
+
+  inline Velocity kiloToVelocity(double f)
+  {
+    Times ts = 3600.0 * boost::units::si::seconds;
+    return kiloToLength(f)/ts;
+  }
+
+  inline Velocity feetToVelocity(double f)
```

Comment:
Feet per hour. This seems a little obscure and probably isn't needed and doesn't appear to be used.